core/src/scene/scene.cpp

```diff
@@ -73,4 +80,9 @@ bool Scene::texture(const std::string& textureName, std::shared_ptr<Texture>& te
     return true;
 }
 
+void Scene::setComponent(std::string componentPath, std::string value) {
+    std::vector<std::string> splitPath = splitString(componentPath, COMPONENT_PATH_DELIMITER);
+    m_userDefinedValues[componentPath] = { splitPath, value };
```

Comment:
Component updates should be stored in a structure that preserves the order of their creation, otherwise we'll end up with update sequences that don't have well-defined outcomes, e.g.
```c++
Tangram::setSceneComponent("scene.animated", "true");
Tangram::setSceneComponent("scene", "null");
Tangram::applySceneUpdates(); // Does this succeed? Depends on your map hash function!
```